src/wagtailtrans/forms.py

```diff
@@ -39,20 +39,24 @@ def __init__(self, *args, **kwargs):
 
         # Remove is_default when a default is set.
         default_language = Language.objects.default()
-        if default_language and (
-                get_wagtailtrans_setting('LANGUAGES_PER_SITE') or
-                get_wagtailtrans_setting('SYNC_TREE')):
+        if ((
+            default_language and
+            get_wagtailtrans_setting('LANGUAGES_PER_SITE') or
+            self.initial.get('is_default', False)
+        )):
             del self.fields['is_default']
 
     def clean_is_default(self):
-        """Force the `is_default` to stay the same, when in sync mode."""
+        is_default = self.cleaned_data['is_default']
         if (
-            self.instance and
-            get_wagtailtrans_setting('SYNC_TREE') and
-            Language.objects.default()
+            not self.initial.get('is_default') == is_default and
+            is_default and
+            not get_wagtailtrans_setting('LANGUAGES_PER_SITE')
         ):
-            return self.instance.is_default
-        return self.cleaned_data['is_default']
+            from wagtailtrans.utils.language_swich import change_default_language  # noqa
+            change_default_language(self.instance)
```

Comment:
Don't think this should be done from within a ``clean_`` function.. Perhaps it's better to do this after saving the object (or at the same time) maybe it's better to implement this in the form save method..

If we're supporting this from within the interface it's also good to have permissions properly checked so not everybody (but only superusers) can change this.